src/main/webapp/src/app/map/map.component.ts

```diff
@@ -11,66 +11,89 @@ import { HiddenGemService } from '../hidden-gem.service';
 })
 
 export class MapComponent implements AfterViewInit {
-
+  
   @ViewChild("mapContainer", { static: false })
   mapContainer!: ElementRef;
   map!: google.maps.Map;
+  hiddenGems = this.appComponent.hiddenGems;
+  type!: string;
   location = {} as Location;
 
-  constructor(private locationService: LocationService) {
-      this.locationService.getLocation
+  constructor( private appComponent: AppComponent, private locationService: LocationService) {
+    this.locationService.getLocation
       .subscribe(location => {
         this.location = location;
-    })
+      })
   }
 
+  ngDoCheck() {
+    if (this.hiddenGems !== this.appComponent.hiddenGems) {
+      this.hiddenGems = this.appComponent.hiddenGems;
+      this.mapInitializer();
+    }
+  }
+  
   ngAfterViewInit(): void {
-    this.mapInitializer();
   }
 
   mapInitializer(): void {
+
     // Coordinates fetched from the user's location to set the center of the map.
     let centerOfMapCoordinates = new google.maps.LatLng(this.location.lat, this.location.lng);
     let mapOptions: google.maps.MapOptions = {
       center: centerOfMapCoordinates,
       zoom: 10
     };
     this.map = new google.maps.Map(this.mapContainer.nativeElement, mapOptions);
+
     this.loadMarkers();
   }
 
   loadMarkers(): void {
-    // Hard-coded dummy markers.
-    const markers = [
-      {
-        position: new google.maps.LatLng( -33.513059, 151.1234),
-        map: this.map,
-        title: "name: Quay restaurant, address: 123 Sydney Street, Sydney, star rating: 3"
-      },
-      {
-        position: new google.maps.LatLng(-33.47,151.865),
-        map: this.map,
-        title: "name: Food Van cafe, address: 123 Sydney Street, Sydney, star rating: 3"
+    // Declare array of markers to keep the fetched data from json string from Hidden Gems object.
+    var markers = new Array();
+
+    var length = this.hiddenGems.length;
+    for (var i = 0; i < length; i++) {
+      
+      // This prevent the tests from failing with "Cannot read property '0' of undefined"
+      if (this.hiddenGems[i].types != null && this.hiddenGems[i].types[0] != null) {
+        this.type = this.hiddenGems[i].types[0];
       }
-    ];
 
+      var data = this.hiddenGems[i],
+      latLng = new google.maps.LatLng(data.geometry.location.lat, data.geometry.location.lng); 
+      //console.log(data.name);
+      // Creating a marker and putting it on the map.
+      const marker = [{
+        position: latLng,
+        map: this.map,
+        title: data.name + data.vicinity +
```

Comment:
Should there be spaces between the strings? also this seems like a really long title.

Reply:
changed title to name of the gem, added spaces in content of infoWindow